src/ansible_compat/config.py

```diff
@@ -0,0 +1,98 @@
+"""Store configuration options as a singleton."""
+import os
+import re
+import subprocess
+import sys
+from argparse import Namespace
+from functools import lru_cache
+from typing import Any, Dict, List, Optional, Tuple
+
+from packaging.version import Version
+
+from ansible_compat.constants import ANSIBLE_MISSING_RC
+
+options = Namespace(
+    cache_dir=None,
+    colored=True,
+    configured=False,
+    cwd=".",
+    verbosity=False,
+    offline=False,
+    project_dir=".",  # default should be valid folder (do not use None here)
+    extra_vars=None,
+)
+
+
+# Used to store collection list paths (with mock paths if needed)
+collection_list: List[str] = []
+
+
+def get_rule_config(rule_id: str) -> Dict[str, Any]:
+    """Get configurations for the rule ``rule_id``."""
+    rule_config = options.rules.get(rule_id, dict())
+    if not isinstance(rule_config, dict):
+        raise RuntimeError("Invalid rule config for %s: %s" % (rule_id, rule_config))
+    return rule_config
+
+
+@lru_cache()
+def ansible_collections_path() -> str:
+    """Return collection path variable for current version of Ansible."""
+    # respect Ansible behavior, which is to load old name if present
+    for env_var in ["ANSIBLE_COLLECTIONS_PATHS", "ANSIBLE_COLLECTIONS_PATH"]:
+        if env_var in os.environ:
+            return env_var
+
+    # https://github.com/ansible/ansible/pull/70007
+    if ansible_version() >= ansible_version("2.10.0.dev0"):
+        return "ANSIBLE_COLLECTIONS_PATH"
+    return "ANSIBLE_COLLECTIONS_PATHS"
+
+
+def parse_ansible_version(stdout: str) -> Tuple[str, Optional[str]]:
+    """Parse output of 'ansible --version'."""
+    # Ansible can produce extra output before displaying version in debug mode.
+
+    # ansible-core 2.11+: 'ansible [core 2.11.3]'
+    match = re.search(r"^ansible \[(?:core|base) ([^\]]+)\]", stdout, re.MULTILINE)
+    if match:
+        return match.group(1), None
```

Comment:
Could you name the group? Having to keep track of numbers adds to the cognitive complexity.